### PR TITLE
Generate Request message input json and validate response to avoid crash

### LINKED ISSUE #148: If response is an object, but mallformed (not having a property value), will crash the dapp
**Describe the bug**
If the response is an object, but malformed (not having a property value), will crash the dapp

### PR DESCRIPTION
**Pull request by tiero**
Closes #148

### CODE REVIEW

**Comment by tiero** [comment]
please review @raamb

**Comment by ferrouswheel** [comment]
Not that the parseServiceSpec is in every UI component, so changes will have to propagate.

There is also a a bug in it, that I had to fix for #156

This is a good highlight for why putting off refactoring the custom UI components #79 is a bad idea. It's going to be impossible to reconcile changes between all the individual components.

File: src/components/service/DefaultService.js [hunk 1]
```diff
@@ -25,14 +30,24 @@ export default class DefaultService extends React.Component {
...
-                    this.state.response = JSON.stringify(nextProps.response);
+                    return
+                }
+                
+                if (nextProps.hasOwnProperty("responseObject") && nextProps.responseObject.hasOwnProperty("value") ) {
+                    this.state.response = nextProps.responseObject.value;
+                    return
                 }
+
+                this.state.response = JSON.stringify(nextProps.response || nextProps.responseObject)
```
Comment (ferrouswheel) [inline src/components/service/DefaultService.js:50]: Please don't stringify the response by default. UI components need access to the underlying structure of the messages and extra serialisation steps makes things frustrating.

Comment (tiero) [reply src/components/service/DefaultService.js:50]: you are right. This was my attempt to fix a crash when there is a malformed response. I will revert the code.

**Comment by ferrouswheel** [comment]
@raamb ... uh, can we wait until the PR is fixed before merging these things?

**Comment by tiero** [comment]
> Not that the parseServiceSpec is in every UI component, so changes will have to propagate.

Do you mean to replace parseServiceSpec with the `protobuf.js` utility? I left to make it backward compatible and not start a refactor and keep this change very isolated.  

> There is also a a bug in it, that I had to fix for #156
I don't understand what I should do 🤔 

> This is a good highlight for why putting off refactoring the custom UI components #79 is a bad idea. It's going to be impossible to reconcile changes between all the individual components.
Agreed

**Comment by ferrouswheel** [comment]
The bug is that we are assigning directly to the component state instead of using `setState`.

But yeah, the whole thing needs to be majorly changed (`componentWillRecieveProps` is deprecated in recent version of react, because people like us misuse it).

**Comment by tiero** [comment]
> The bug is that we are assigning directly to the component state instead of using `setState`.

Yes at first glance even to me that was very strange, but if it was in master I assumed was ok or for some reason needed. (Isn't very common? 😂) 

~~Will fix in #164~~

**Comment by raamb** [comment]
Guys. I am working with a react expert to come up with a workable
solution.  Should have it in a day or two. We are trying  to resolve this
via composition rather than inheritance.

Thanks,
Raam

**Pull request merged**